runner/runner.go

```diff
@@ -169,10 +168,19 @@ func (r *Runner) Run(ctx context.Context, args []string) int {
 
 	// Start stats
 	// TODO: remove as soon as we update the configuration with statsExcludedTags where necessary
-	if !config.IsSet("statsExcludedTags") && deploymentType == deployment.MultiTenantType && (!config.IsSet("WORKSPACE_NAMESPACE") || strings.Contains(config.GetString("WORKSPACE_NAMESPACE", ""), "free")) {
+	if !config.IsSet("statsExcludedTags") && deploymentType == deployment.MultiTenantType &&
+		(!config.IsSet("WORKSPACE_NAMESPACE") || strings.Contains(config.GetString("WORKSPACE_NAMESPACE", ""), "free")) {
 		config.Set("statsExcludedTags", []string{"workspaceId", "sourceID", "destId"})
 	}
-	stats.Default.Start(ctx)
+	stats.Default = stats.NewStats(config.Default, logger.Default, svcMetric.Instance,
+		stats.WithServiceName(r.appType),
+		stats.WithServiceVersion(r.releaseInfo.Version),
+		stats.WithNamespace(config.GetNamespaceIdentifier()),
+		stats.WithInstanceName(config.GetString("INSTANCE_ID", "")),
+	)
+	if err := stats.Default.Start(ctx); err != nil {
```

Comment:
There is a test for it i.e. `TestOTelStartStopError` in `/services/stats/otel_test.go`. 
In the test I'm not providing an endpoint which makes `Start()` return an error and then I invoke `Stop()` to see if it blocks or not. Let me know if you think more cases should be added.

Reply:
In case of an error does it retry to connect behind-the-scenes? If not, we should return and not start the server. Otherwise, it is fine to simply log an error and resume.

Reply:
If it can't connect an error is not returned unless we block but we moved away from that implementation. That's how it works by design (see `grpc.DialContext`). This is a double edged-sword since we might not know if we managed to connect after all (maybe we pass the wrong port and it's not possible to establish a connection ever).

In order to mitigate this we are setting up alerts that will warn us if no metrics are being received. The collectors expose several useful metrics such as accepted metrics, refused metrics, etcetera that I think should be a good enough indicator of the health of our monitoring stack. At least for now.

If this will prove not to be sufficient, an alternative would be to manage the lifecycle of the connection ourselves, but before going down that road and burden us with the additional complexity, I think this apprach should be reliable enough and similar to what we already have in place.

On the other end, if an error is returned, it is because we passed something really wrong in terms of configuration (e.g. you enable OpenTelemetry but then you don't provide any metrics nor traces endpoint) so it might make sense to return. Configuration wise I never managed to get an actual error from the underlying OTel framework though but I think the same should apply there.

I'll make the change to the regulation worker as well :+1: